tfhe/src/integer/oprf.rs

```diff
@@ -0,0 +1,175 @@
+
+        let seeds: Vec<Seed> = (0..num_blocks)
+            .map(|_| deterministic_seeder.seed())
+            .collect();
+
+        let blocks = seeds
+            .into_par_iter()
+            .enumerate()
+            .map(|(i, seed)| {
+                let i = i as u64;
+
+                if i * message_bits_count < random_bits_count {
+                    // if we generate 5 bits of noise in n blocks of 2 bits, the third (i=2) block
+                    // must have only one bit of random
+                    if random_bits_count < (i + 1) * message_bits_count {
+                        let top_message_bits_count = random_bits_count - i * message_bits_count;
+
+                        assert!(top_message_bits_count <= message_bits_count);
+
+                        self.key
+                            .generate_oblivious_pseudo_random(seed, top_message_bits_count)
+                    } else {
+                        self.key
+                            .generate_oblivious_pseudo_random(seed, message_bits_count)
+                    }
+                } else {
+                    self.key.create_trivial(0)
+                }
+            })
+            .collect::<Vec<_>>();
+
+        RadixCiphertext::from(blocks)
+    }
+}
+
+// Describes in which range a random signed integer should be generated
+#[derive(Copy, Clone)]
+pub enum SignedRandomizationSpec {
+    // taken uniformly in `[0, 2^random_bits_count[`
+    Unsigned { random_bits_count: u64 },
+    // taken uniformly in the full range [-2^(p-1), 2^(p-1)[
+    FullSigned,
+}
+
+impl ServerKey {
+    /// Generates an encrypted `num_block` blocks signed integer
+    /// using the given seed following the randomizer spec
+    /// The encryted value is oblivious to the server
+    /// It can be useful to make server random generation deterministic
+    ///
+    /// ```rust
+    /// use tfhe::integer::gen_keys_radix;
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    /// use tfhe::Seed;
+    ///
+    /// let size = 4;
+    ///
+    /// // Generate the client key and the server key:
+    /// let (cks, sks) = gen_keys_radix(PARAM_MESSAGE_2_CARRY_2_KS_PBS, size);
+    ///
+    /// let random_bits_count = 3;
+    ///
+    /// let ct_res = sks.par_generate_oblivious_pseudo_random_signed_integer(
+    ///     Seed(0),
+    ///     SignedRandomizationSpec::Unsigned { random_bits_count },
+    ///     size as u64,
+    /// );
+    ///
+    /// // Decrypt:
+    /// let dec_result = cks.decrypt(&ct_res);
+    /// assert!(dec_result < (1 << random_bits_count));
+    /// ```
+    pub fn par_generate_oblivious_pseudo_random_signed_integer(
+        &self,
+        seed: Seed,
+        randomizer: SignedRandomizationSpec,
+        num_blocks: u64,
+    ) -> RadixCiphertext {
+        if let SignedRandomizationSpec::Unsigned { random_bits_count } = randomizer {
+            assert!(2 * (1 << random_bits_count) <= num_blocks * self.message_modulus().0 as u64);
```

Comment:
let's have some error message for the asserts please (al asserts for this function and the one above),, so that users can know why something went wrong in more human terms, can be a follow up with the HL API, unless you plan on adding the HL stuff in this PR as well